include/mbedtls/fsio.h

```diff
@@ -0,0 +1,244 @@
+
+#else /* !MBEDTLS_FS_IO_ALT */
+
+/**
+ * file context.
+ *
+ */
+typedef void mbedtls_file_t;
+
+/**
+ * dir context.
+ *
+ */
+typedef void mbedtls_dir_t;
+
+/**
+ * Definition of whence required by mbedtls_fseek().
+ */
+#define MBEDTLS_SEEK_SET    0
+#define MBEDTLS_SEEK_CUR    1
+#define MBEDTLS_SEEK_END    2
+
+/**
+ * Definition of dir entry types required by mbedtls_readdir().
+ */
+#define MBEDTLS_FSIO_DT_BLK     0
+#define MBEDTLS_FSIO_DT_CHR     1
+#define MBEDTLS_FSIO_DT_DIR     2
+#define MBEDTLS_FSIO_DT_FIFO    3
+#define MBEDTLS_FSIO_DT_LNK     4
+#define MBEDTLS_FSIO_DT_REG     5
+#define MBEDTLS_FSIO_DT_SOCK    6
+#define MBEDTLS_FSIO_DT_UNKNOWN 7
+
+/**
+ * \brief          Open file. Follows standard C fopen interface.
+ *
+ * \param path     File path
+ * \param mode     Open mode
+ *
+ * \return         Pointer to mbedtls_file_t on success or NULL on failure.
+ */
+mbedtls_file_t * mbedtls_fopen( const char *path, const char *mode );
+
+/**
+ * \brief          Read file. Follows standard C fread interface.
+ *
+ * \param ptr      Pointer to output buffer
+ * \param size     Size of read items.
+ * \param nmemb    Number of read items.
+ * \param stream   Pointer to mbedtls_file_t.
+ *
+ * \return         Number of items read.
+ */
+size_t mbedtls_fread( void *ptr, size_t size, size_t nmemb,
+                      mbedtls_file_t *stream );
+
+/**
+ * \brief          Write file. Follows standard C fwrite interface.
+ *
+ * \param ptr      Pointer to input buffer
+ * \param size     Size of write items.
+ * \param nmemb    Number of write items.
+ * \param stream   Pointer to mbedtls_file_t.
+ *
+ * \return         Number of items written.
+ */
+size_t mbedtls_fwrite( const void *ptr, size_t size, size_t nmemb,
+                       mbedtls_file_t *stream );
+
+/**
+ * \brief          Reads a line from file. Follows standard C fgets interface.
+ *
+ * \param s        Pointer to output buffer.
+ * \param size     Size of buffer.
+ * \param stream   Pointer to mbedtls_file_t.
+ *
+ * \return         returns s on success, and NULL on error or
+ *                 when end of file occurs while no characters have been read.
+ */
+char * mbedtls_fgets( char *s, int size, mbedtls_file_t *stream );
+
+/**
+ * \brief          Sets file position. Follows standard C fseek interface.
+ *
+ * \param stream   Pointer to mbedtls_file_t.
+ * \param offset   Offset from whence.
+ * \param whence   Position from where offset is applied.
+ *                 Value is one of MBEDTLS_SEEK_SET, MBEDTLS_SEEK_CUR, or MBEDTLS_SEEK_END.
+ *
+ * \return         returns 0 on success, and -1 on error
+ */
+int mbedtls_fseek( mbedtls_file_t *stream, long offset, int whence );
+
+/**
+ * \brief          Gives current position of file in bytes.
+ *                 Follows standard C ftell interface.
+ *
+ * \param stream   Pointer to mbedtls_file_t.
+ *
+ * \return         returns current position on success, and -1 on error
+ */
```

Comment:
Perhaps elaborate by saying that the position is measured in bytes from the beginning of the file?